wgpu/src/backend/direct.rs

```diff
@@ -2409,19 +2424,20 @@ pub struct QueueWriteBuffer {
     mapping: BufferMappedRange,
 }
 
-impl std::ops::Deref for QueueWriteBuffer {
-    type Target = [u8];
-
-    fn deref(&self) -> &Self::Target {
-        panic!("QueueWriteBuffer is write-only!");
+impl crate::context::QueueWriteBuffer for QueueWriteBuffer {
+    fn slice(&self) -> &[u8] {
+        use crate::BufferMappedRangeSlice;
+        self.mapping.slice()
```

Comment:
This should still panic

Reply:
Just a question, you can still read from the mut slice though?

Reply:
Yes. It's just to discourage it as it may be slow as hell.